### PR TITLE
Improve Travel Plan Score Calculation

### PR DESCRIPTION
**Pull request by rwangsc18**
## Description
Clearly describe changes in your PR, new feature or bug fixes
Update the plan score function for efforts to improve plan generation quality. Its [design doc](https://docs.google.com/document/d/1XNIJ9nKIoW6SnmV3AV4vzgy2vbR7fiVA2ywqXhVZ02w/edit) is shared.

## Root Cause
Briefly describe the root cause and analysis of the problem
Current plan score function can favor long travel distance in certain scenarios, details are disclosed in the [design doc](https://docs.google.com/document/d/1XNIJ9nKIoW6SnmV3AV4vzgy2vbR7fiVA2ywqXhVZ02w/edit)

## Solution
Describe your code changes in detail for reviewers. Explain the solution and how it fixes the issue.
For new features, describe the high-level ideas.
Use a constant distance normalization factor (e.g. search radius). In the future, we plan to allow user to adjust this distance normalization factor. 

## Testing
Did you add any new test for this change, performed manual integration tests, or both?
Added some side-by-side comparison results in the design doc. Updated one existing unit test case, and added one new unit test. In the future, we plan to add more related with the distance penalty score.

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt to format your code ?
- [x] You have added unit tests

### CODE REVIEW

**Review by weihesdlegend — APPROVED** [review]
LGTM, we should add TODOs to remove the `ProdScore` once we are happy with the new formula.

File: matching/score.go [hunk 1]
```diff
@@ -20,7 +21,17 @@ func Score(places []Place) float64 {
 	distances := calDistances(places)                     // Haversine distances
 	maxDist := math.Max(0.001, calMaxDistance(distances)) // protect against maximum distance being zero
 	avgDistance := stat.Mean(distances, nil) / maxDist    // normalized average distance
+	avgScore := avgPlacesScore(places)
+
+	return avgScore - avgDistance
```
Comment (sananand007) [inline matching/score.go:26]: we are not worried about absolute values rite ? Negative values are also ok (as earlier)?

**Review by sananand007 — APPROVED** [review]
LGTM, just an [nit] on the difference calculation.

File: matching/score.go [hunk 2]
```diff
@@ -20,7 +21,17 @@ func Score(places []Place) float64 {
...
+}
 
+func ScoreNew(places []Place, distNormConst int) float64 {
```
[the remark this answers is not in the document]
Comment (rwangsc18) [reply matching/score.go]: Added a short comment

[the remark this answers is not in the document]
Comment (rwangsc18) [reply matching/score.go]: Done

File: matching/score.go (hunk 1 above)
Comment (rwangsc18) [reply matching/score.go:28]: we are okay with negative values. When the final score is negative, it just means the penalty from distance is greater than the positive gains of plan locations themselves. 
Sry about the late reply, I just saw your comments.

**Pull request merged**